python/pyiceberg/catalog/__init__.py

```diff
@@ -137,6 +145,10 @@ def infer_catalog_type(name: str, catalog_properties: RecursiveDict) -> Optional
                 return CatalogType.REST
             elif uri.startswith("thrift"):
                 return CatalogType.HIVE
+            elif uri.startswith("postgresql"):
+                return CatalogType.JDBC
+            elif uri.startswith("file"):
```

Comment:
I'm not sure this makes sense to me. Why would a "file://" catalog URI signal that a database should be used?

Reply:
I'll remove this, it's a left over of when I tried to support both PostgreSQL and sqlite.  I quickly realized that abstracting the database driver would be more difficult that I thought without relying on a framework like SQLAlchemy.  Not impossible, but maybe a little out of scope for this initial draft.